src/libre/re_comp.c

```diff
@@ -24,6 +57,77 @@ comp_iter_repeated(struct comp_env *env,
     struct fsm_state *x, struct fsm_state *y,
     struct ast_expr_repeated *n);
 
+/* Return the start/end any loop, creating if necessary. */
+static struct fsm_state *
+intern_start_any_loop(struct comp_env *env);
+static struct fsm_state *
+intern_end_any_loop(struct comp_env *env);
+
+static int
+comp_iter(struct comp_env *env,
+    struct fsm_state *x, struct fsm_state *y,
+    struct ast_expr *n);
+
+static int
+can_skip_concat_state_and_epsilon(const struct ast_expr *l,
+    const struct ast_expr *r);
+
+static struct fsm *
+new_blank(const struct fsm_options *opt);
```

Comment:
Is this the same as `fsm_empty()`?

Reply:
`fsm_empty` returns whether the fsm is empty (it's a predicate), this returns a new FSM with a start state.